metrics.py

```diff
@@ -0,0 +1,196 @@
+            grayscale = rgb2gray(image)
+            edges = canny(grayscale, sigma=3.0)
+            out, angles, distances = hough_line(edges)
+            _, angles_peaks, _ = hough_line_peaks(out, angles, distances, num_peaks=20)
+            angle=np.mean(np.rad2deg(angles_peaks))
+            if 0 <= angle <= 90:
+                rot_angle = angle - 90
+            elif -45 <= angle < 0:
+                rot_angle = angle - 90
+            elif -90 <= angle < -45:
+                rot_angle = 90 + angle
+            if abs(rot_angle)>20:
+                rot_angle=0
+            rotated = rotate(image, rot_angle, resize=True)*255
+            rotated =rotated.astype(np.uint8)
+            rotated1=rotated[:,:,:]
+            minus=np.abs(int(np.sin(np.radians(rot_angle))*rotated.shape[0]))
+            if rotated.shape[1]/rotated.shape[0]<2 and minus >10:
+
+                rotated1=rotated[minus:-minus,:,:]
+            lab= cv2.cvtColor(rotated1, cv2.COLOR_BGR2LAB)
+            l, a, b = cv2.split(lab)
+            clahe = cv2.createCLAHE(clipLimit=3.0, tileGridSize=(8,8))
+            cl = clahe.apply(l)
+            limg = cv2.merge((cl,a,b))
+            final = cv2.cvtColor(limg, cv2.COLOR_LAB2BGR)
+            paths='./model1_nomer.tflite'
+            interpreter = tf.lite.Interpreter(model_path=paths)
+            interpreter.allocate_tensors()
+            # Get input and output tensors.
+            input_details = interpreter.get_input_details()
+            output_details = interpreter.get_output_details()
+    #         img =rotated1
+            img =final # лучше работает при плохом освещении
+
+            img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+            img = cv2.resize(img, (128,64))
+            img = img.astype(np.float32)
+            img /= 255
+
+            img1=img.T
+            img1.shape
+            X_data1=np.float32(img1.reshape(1,128, 64,1))
+            input_index = (interpreter.get_input_details()[0]['index'])
+            interpreter.set_tensor(input_details[0]['index'], X_data1)
+
+            interpreter.invoke()
+
+            net_out_value = interpreter.get_tensor(output_details[0]['index'])
+            pred_text = decode_batch(net_out_value)
+            elapsed_time.append((datetime.now()-start_time).total_seconds())
+            result[true_number]=pred_text
+        else :
+            pred_texts = ''
+            result[true_number]=pred_text
+            elapsed_time.append((datetime.now()-start_time).total_seconds())
+        bar()
+    
+# Расчет необходимых метрик
+latency=np.mean(elapsed_time)
+lev_distance=[levenstein(i[0],i[1][0]) for i in result.items()]
+true_positive=sum([len(i[0]) - levenstein(i[0],i[1][0]) for i in result.items()])
+precision=true_positive/sum([len(i[0]) for i in result.values()])
+recall=true_positive/sum([len(i) for i in result.keys()])
+ocr_total=len(number)
+ocr_error=sum([1 if levenstein(i[0],i[1][0]) >=3 else 0 for i in result.items()])
```

Comment:
3 - это пример "магической константы" в коде. Как минимум, нужно в начале скрипта сделать раздел с глобальными переменными, и ввести там что-то вроде LEVENSTEIN_THRESHOLD = 3

Тогда даже код

```suggestion
ocr_error=sum([1 if levenstein(i[0],i[1][0]) >= LEVENSTEIN_THRESHOLD else 0 for i in result.items()])
```

станет более читаемым.

Ещё более правильно вынести это в отдельный файл вроде test_config.py, и делать import test_config
Для интеграции с dvc, возможно, полезно вынести это в файл yaml-параметров, но тут вопрос спорный, т.к. к процессу обучения порог на число правильных символов не имеет отношения, по крайней мере, пока. Правда, на результат в файле метрик очень даже влияет.